src/unix/apple/audio.c

```diff
@@ -4,22 +4,28 @@
 
 #include "matoya.h"
 
+#include <CoreAudio/CoreAudio.h>
 #include <AudioToolbox/AudioToolbox.h>
 
-#define AUDIO_SAMPLE_SIZE sizeof(int16_t)
+#define AUDIO_SAMPLE_SIZE(fmt) ((fmt) == MTY_AUDIO_SAMPLE_FORMAT_FLOAT ? sizeof(float) : sizeof(int16_t))
 #define AUDIO_BUFS        64
 
 #define AUDIO_BUF_SIZE(ctx) \
-	((ctx)->sample_rate * (ctx)->channels * AUDIO_SAMPLE_SIZE)
+	((ctx)->sample_rate * (ctx)->channels * AUDIO_SAMPLE_SIZE((ctx)->sample_format))
+
+#define AUDIO_HW_ERROR(e) ((e) != kAudioHardwareNoError)
+#define AUDIO_SV_ERROR(e) ((e) != kAudioServicesNoError)
 
 struct MTY_Audio {
 	AudioQueueRef q;
 	AudioQueueBufferRef audio_buf[AUDIO_BUFS];
 	MTY_Atomic32 in_use[AUDIO_BUFS];
+	MTY_AudioSampleFormat sample_format;
 	uint32_t sample_rate;
```

Comment:
Add sample size here, so it just needs to be evaluated once.
```suggestion
	MTY_AudioSampleFormat sample_format;
	size_t sample_size;
	uint32_t sample_rate;
```

Where the it will be: `sample_size  = channels * AUDIO_SAMPLE_SIZE(ctx->sample_format)`

Reply:
Done, let me know if you're happy with the changes.

Reply:
I assume this ended up as the `frame_size` member I see